autoparallel/collective_runtime_estimation.py

```diff
@@ -70,10 +71,16 @@ def redistribute_cost(
     # 3. allreduce 4. reduce_scatter
     curr_placements = [current_spec.placements[i] for i in order]
     tgt_placements = [target_spec.placements[i] for i in order]
+    # TODO: is there a better way of doing this that doesn't involve making a tensor?
+    is_contiguous: bool = torch.empty_strided(
+        current_spec.shape, current_spec.stride, device="meta"
+    ).is_contiguous()
```

Comment:
is there a helper function that takes sizes / strides and computes if the tensor is contiguous? It feels like an overkill to have to construct a tensor for this

Reply:
yep, we have `check_contiguous_sizes_strides` (also handles dynamic shapes): https://github.com/pytorch/pytorch/blob/main/torch/_prims_common/__init__.py#L252